app/src/main/java/edu/uwstout/p2pchat/ExpandableSettingsListViewAdapter.java

```diff
@@ -0,0 +1,214 @@
+     * Class is responsible for intepretting the
+     * data in a cleaner way.
+     */
+    private SettingNode views;
+
+    /**
+     * The context that will be used for initializing and changing
+     * fragments.
+     */
+    private Context classContext;
+
+
+
+    /**
+     * Constructor for building an expandable list view adapter.
+     * Initializes the name arrays into views.
+     *
+     * @param context
+     *         the context of the class.
+     */
+    public ExpandableSettingsListViewAdapter(final Context context, Activity currentActivity,
+            FragmentManager manager)
+    {
+        // Assign the context.
+        classContext = context;
+
+
+
+
+        //Assign all texts of childviews the following.
+        String[][] childNames = {
+                {
+                        "Creator: ", "Device Address: ",
+                        "Device Name: ",
+                        "Primary Device Type: ", "Secondary Device Type: ", "Status: "
+                }, {
+                "Light",
+                "Majestic", "Dark"
+        }, {}, {}
+        };
+
+
+        views = new SettingNode(context, childNames, currentActivity, manager);
+
+
+
+    }
+
+    /**
+     * Gets count of parent nodes.
+     * @return
+     */
+    @Override
+    public int getGroupCount()
+    {
+        return views.getParent().length;
+    }
+
+    /**
+     * Gets the count of each child under a certain parent.
+     * @param parent specify parent position.
+     * @return number of children under a certain view.
+     */
+    @Override
+    public int getChildrenCount(final int parent)
+    {
+        //Returns the child size.
+        return views.getChildLength(parent);
+    }
+
+    /**
+     * @param i parent index.
+     * @return string of the parents.
+     */
+    @Override
+    public Object getGroup(final int i)
+    {
+        return views.getParent()[i].getText().toString();
+    }
+
+    /**
+     * Get the text from a child.
+     * @param i parent index.
+     * @param i1 child index;
+     * @return the text of the child.
+     */
+    @Override
+    public Object getChild(final int i, final int i1)
+    {
+        return views.getView(i, i1).getText().toString();
+    }
+
+    /**
+     * Gets the group id of the parent.
+     * @param i position of parent.
+     * @return parent location from i.
+     */
+    @Override
+    public long getGroupId(final int i)
+    {
+
+        return i;
+    }
+
+    /**
+     * Get the position of child given the parent index.
+     * @param generalSetting parent index
+     * @param specificSetting specific child index to look at.
+     * @return the child position.
+     */
+    @Override
+    public long getChildId(final int generalSetting, final int specificSetting)
+    {
+        return specificSetting;
+    }
+
+    /**
+     * Will be false.
+     * @return false.
+     */
+    @Override
+    public boolean hasStableIds()
+    {
+        return false;
```

Comment:
hard coded method, not good. I can see that this is an override, so if there is a reason that you have hard-coded this method, you need to explain that in the documentation.